nf_core/lint_utils.py

```diff
@@ -60,12 +63,55 @@ def run_prettier_on_file(file):
         If Prettier is not installed, a warning is logged.
     """
 
+    if shutil.which("prettier"):
+        _run_prettier_on_file(file)
+    elif shutil.which("pre-commit"):
+        _run_pre_commit_prettier_on_file(file)
+    else:
+        log.warning(
+            "Neither Prettier nor the prettier pre-commit hook are available. At least one of them is required."
+        )
+
+
+def _run_prettier_on_file(file):
+    """Run natively installed Prettier on a file."""
+
     try:
         subprocess.run(
             ["prettier", "--write", file],
-            stdout=subprocess.DEVNULL,
-            stderr=subprocess.DEVNULL,
-            check=False,
+            capture_output=True,
+            check=True,
+        )
+    except subprocess.CalledProcessError as e:
+        if ": SyntaxError: " in e.stderr.decode():
+            raise ValueError(f"Can't format {file} because it has a synthax error.\n{e.stderr.decode()}") from e
+        raise ValueError(
+            "There was an error running the prettier pre-commit hook.\n"
+            f"STDOUT: {e.stdout.decode()}\nSTDERR: {e.stderr.decode()}"
+        ) from e
+
+
+def _run_pre_commit_prettier_on_file(file):
+    """Runs pre-commit hook prettier on a file if pre-commit is installed.
+
+    Args:
+        file (Path | str): A file identifier as a string or pathlib.Path.
+
+    Warns:
+        If Prettier is not installed, a warning is logged.
+    """
+
+    nf_core_pre_commit_config = Path(nf_core.__file__).parent.parent / ".pre-commit-config.yaml"
+    try:
+        subprocess.run(
+            ["pre-commit", "run", "--config", nf_core_pre_commit_config, "prettier", "--files", file],
+            capture_output=True,
+            check=True,
         )
```

Comment:
Just wondering if we can import pre-commit the package and call this manually? Or is using it as a subprocess more transparent.

Reply:
I think the manual and the subprocess version serve different purposes. The automated one should make sure all autogenerated file changes are lint-compliant and the manual invocations can help after manually changing files plus installing the pre-commit hooks will protect the git revision history from content that doesn't pass linting.

Reply:
Sorry, answered the wrong question. The real one I don't know. Well, I kind of know. It's not the intended use of pre-commit. Not impossible, but a hack.

Reply:
If you want to know why I classify as a "hack", compare the amount and complexity of the parameters in the [`run`](https://github.com/pre-commit/pre-commit/blob/1f59f4cba8241ed7e572a1b5818bba74c6f31181/tests/commands/run_test.py#L140) function wit the much simpler command line invocation.

Reply:
On a sidenote: running a command in subprocess adds very little overhead (about 0.04 seconds on a 10 year old machine) and there's no state or computation that we already have when we invoke it that we can share with prettier to make it run faster anyway. I would say we can drop this line of inquiry.